app/docker/views/networks/create/createNetworkController.js

```diff
@@ -1,138 +1,195 @@
+      function prepareLabelsConfig(config) {
+        config.Labels = LabelHelper.fromKeyValueToLabelHash($scope.formValues.Labels);
+      }
 
-    var nodeName = $scope.formValues.NodeName;
-    HttpRequestHelper.setPortainerAgentTargetHeader(nodeName);
-
-    $scope.state.actionInProgress = true;
-    NetworkService.create(networkConfiguration)
-    .then(function success(data) {
-      var networkIdentifier = data.Id;
-      var userId = userDetails.ID;
-      return ResourceControlService.applyResourceControl('network', networkIdentifier, userId, accessControlData, []);
-    })
-    .then(function success() {
-      Notifications.success('Network successfully created');
-      $state.go('docker.networks', {}, {reload: true});
-    })
-    .catch(function error(err) {
-      Notifications.error('Failure', err, 'An error occured during network creation');
-    })
-    .finally(function final() {
-      $scope.state.actionInProgress = false;
-    });
-  };
-
-  function initView() {
-    var apiVersion = $scope.applicationState.endpoint.apiVersion;
-
-    PluginService.networkPlugins(apiVersion < 1.25)
-    .then(function success(data){
-        $scope.availableNetworkDrivers = data;
-    })
-    .catch(function error(err) {
-      Notifications.error('Failure', err, 'Unable to retrieve network drivers');
-    });
-  }
-
-  initView();
-}]);
+      function prepareConfiguration() {
+        var config = angular.copy($scope.config);
+        prepareIPAMConfiguration(config);
+        prepareDriverOptions(config);
+        prepareLabelsConfig(config);
+        return config;
+      }
+
+      function modifyNetworkConfigurationForMacvlanConfigOnly(config) {
+        config.ConfigOnly = true;
+        config.Options.parent = $scope.formValues.Macvlan.ParentNetworkCard;
+      }
+
+      function modifyNetworkConfigurationForMacvlanConfigFrom(config, selectedNetworkConfig) {
+        config.ConfigFrom = {
+          Network: selectedNetworkConfig.Name
+        };
+        config.Scope = 'swarm';
+      }
+
+      function validateForm(accessControlData, isAdmin) {
+        $scope.state.formValidationError = '';
+        var error = '';
+        error = FormValidator.validateAccessControl(accessControlData, isAdmin);
+
+        if (error) {
+          $scope.state.formValidationError = error;
+          return false;
+        }
+        return true;
+      }
+
+      function createNetwork(networkConfiguration, userDetails, accessControlData, reload) {
+        $scope.state.actionInProgress = true;
+        NetworkService.create(networkConfiguration)
+          .then(function success(data) {
+            var networkIdentifier = data.Id;
+            var userId = userDetails.ID;
+            return ResourceControlService.applyResourceControl('network', networkIdentifier, userId, accessControlData, []);
+          })
+          .then(function success() {
+            Notifications.success('Network successfully created');
+            if (reload)
```

Comment:
I'm not sure to understand when we need to reload and when not?

Reply:
We can deploy the macvlan config network to many nodes, but we can't create it on all nodes at the same time, we should not redirect the app to `/networks` until the last network config is sent to the API.
So we reload only after the last call of config network creation, but we always reload on all cases